content-scope-scripts/content-scope-scripts-impl/src/main/java/com/duckduckgo/contentscopescripts/impl/RealContentScopeScripts.kt

```diff
@@ -34,6 +34,7 @@ import javax.inject.Inject
 
 interface CoreContentScopeScripts {
     fun getScript(): String
+    fun isEnabled(): Boolean
```

Comment:
I have mixed feelings here. From one side we don't really need the `isEnabled()` interface in more places because we already have `contentScopeScriptsFeature.self().isEnabled()` which is the same thing, so this is a pass through.

On the other hand, not having this interface means adding an extra dependency in places like `BrowserWebClient`

I think we need to change the API and make this an implementation detail of it.

Rather than having `fun getScript(): String` in the `ContentScopeScripts` API. I think we should change it to `fun `injectContentScopeScripts(webview: WebView)` or similar and move all the logic to check if the toggle is enabled or not inside the implementation. That way we wouldn't return the script but rather directly inject it (or not) in the given webview instance.